openupgrade_scripts/scripts/account/17.0.1.2/post-migration.py

```diff
@@ -0,0 +1,535 @@
+    you have to create yourself beforehand.
+
+    :param model_name: Name of the model.
+    :param origin_field_name: Name of the company-dependent field
+    :param destination_field_name: Name of plain field
+    :param origin_id_column_name: Name of the column you created to link record
+      duplicates to the record they were duplicated from
+    :param model_table_name: Name of the table. Optional. If not provided
+      the table name is taken from the model (so the model must be
+      registered previously).
+    """
+    # If you want to recycle this function for your own migration, better
+    # add it to openupgradelib
+    table_name = model_table_name or env[model_name]._table
+
+    env.cr.execute(
+        "SELECT id, ttype FROM ir_model_fields "
+        f"WHERE model='{model_name}' AND name='{origin_field_name}'"
+    )
+    field_id, field_type = env.cr.fetchone()
+
+    value_expression = ir_property_TYPE2FIELD.get(field_type)
+    if value_expression == "value_reference":
+        value_expression = "SPLIT_PART(ip.value_reference, ',', 2)::integer"
+
+    return openupgrade.logged_query(
+        env.cr,
+        f"""
+        UPDATE {table_name}
+        SET {destination_field_name} = (
+            SELECT {value_expression}
+            FROM ir_property ip
+            WHERE ip.fields_id={field_id} --- {origin_field_name}
+            AND (
+                ip.res_id = '{model_name}.' || COALESCE(
+                    {table_name}.{origin_id_column_name}, {table_name}.id
+                )
+                OR ip.res_id IS NULL
+            )
+            AND (
+                ip.company_id = {table_name}.company_id
+                OR ip.company_id IS NULL
+            )
+            ORDER BY ip.company_id NULLS LAST, ip.res_id NULLS LAST
+            LIMIT 1
+        )
+        """,
+    )
+
+
+def _account_tax_group_migration(env):
+    """
+    In v17 tax groups are company-aware (company_id added):
+
+    - Find which tax groups have accounts with different companies,
+      duplicate them for each of that company
+    - Update accounts to point to the newly created groups for their
+      companies
+    - Rename ir model data (xml_id), the format will be
+      "{module_name}.{company_id}_xml_id"
+    - Fill new fields tax_receivable_account_id, tax_payable_account_id,
+      advance_tax_payment_account_id with the value of the properties
+      they replace
+
+    Example in v16:
+    2 VN CoA company: tax 0, tax 5, tax 10
+    2 Generic CoA company tax 15
+    1 Belgium CoA company tax 6, 12, 21
+
+    -> After migration we will have 2 tax 0, 2 tax 5, 2 tax 10
+    and 2 tax 15 of course with only different company_id
+    Also the new one will have their own xml_id using create method
+    of ir.model.data
+    And then in each l10n module, only need to perform rename xml_id like
+    https://github.com/Viindoo/OpenUpgrade/pull/655
+    """
+    origin_id_column = openupgrade.get_legacy_name("origin_id")
```

Comment:
```suggestion
    origin_id_column = "original_tax_group_id"

```
Using "origin_id" seems like a field was named origin_id at some point. Also, the use of get_legacy_name() encourages to think this way. To avoid this, I prefer a more meaningful name, more related to tax groups.